app/main.py

```diff
@@ -1,2 +1,55 @@
+import dataclasses
+import math
+from typing import Any
+
+
+@dataclasses.dataclass
+class Node:
+    key: Any
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self):
+        self.size = 0
+        self.capacity = 8
+        self.data = [None] * self.capacity
+
+    def __setitem__(self, key, value):
+
+        index = hash(key) % self.capacity
+
+        while self.data[index] is not None:
+            if self.data[index].key == key:
+                self.data[index].value = value
+                return
+
+            index = (index + 1) % self.capacity
+
+        if self.size >= math.floor(self.capacity * 2 // 3):
+            self._resize_data()
+            self.__setitem__(key=key, value=value)
+            return
+
+        self.size += 1
+        self.data[index] = Node(key=key, value=value)
+
+    def __getitem__(self, key):
+        index = hash(key) % self.capacity
+        while self.data[index] is not None:
+            if self.data[index].key == key:
+                return self.data[index].value
+            index = (index + 1) % self.capacity
+        raise KeyError
+
+    def _resize_data(self):
+        self.capacity *= 2
+        items = [item for item in self.data if item is not None]
+        self.size = 0
+        new_data = [None] * self.capacity
+        self.data = new_data
```

Comment:
I think no need new_data variable
```suggestion
        self.data = [None] * self.capacity
```